httpd/handler.go

```diff
@@ -350,27 +349,15 @@ func httpError(w http.ResponseWriter, error string, code int) {
 // as basic auth: http://username:password@127.0.0.1
 func parseCredentials(r *http.Request) (string, string, error) {
 	q := r.URL.Query()
-	username, password := q.Get("u"), q.Get("p")
-	if username != "" && password != "" {
-		return username, password, nil
-	}
-	auth := r.Header.Get("Authorization")
-	if auth == "" {
-		return "", "", nil
-	}
-	fields := strings.Split(auth, " ")
-	if len(fields) != 2 {
-		return "", "", fmt.Errorf("invalid Basic Authentication header")
-	}
-	bs, err := base64.StdEncoding.DecodeString(fields[1])
-	if err != nil {
-		return "", "", fmt.Errorf("invalid Base64 encoding")
+
+	if u, p := q.Get("u"), q.Get("p"); u != "" && p != "" {
+		return u, p, nil
 	}
-	fields = strings.Split(string(bs), ":")
-	if len(fields) != 2 {
-		return "", "", fmt.Errorf("invalid Basic Authentication value")
+	if u, p, ok := r.BasicAuth(); ok {
+		return u, p, nil
+	} else {
+		return "", "", fmt.Errorf("unable to parse username/password from header")
```

Comment:
Minor: could you make this message more precise e.g. "unable  to parse Basic Auth credentials"?